### PR TITLE
Add vectors for unkeyed blake2s and blake2b.

### PR DESCRIPTION
**Pull request by Ayrx**
OpenSSL [recently added][0] support for the Blake2 hash and it should be in the 1.1.0 release. This PR adds the test vectors for 2 variants of the Blake2 hash.

The vectors were generated as part of that OpenSSL commit per the process [described in the discussions][1]. The main reason being that the vectors found in the Blake2 [repository][2] are for key != 0 (more akin to a MAC) while the OpenSSL function always set key == 0.

If / when this gets merged I'll put out a PR for the actual implementation targeting to have it available and merge ready when we add support for 1.1.0 and there's an official release out.

[0]: https://github.com/openssl/openssl/commit/2d0b44126763f989a4cbffbffe9d0c7518158bb7
[1]: https://github.com/openssl/openssl/pull/566#discussion-diff-55609025
[2]: https://github.com/BLAKE2/BLAKE2/tree/master/testvectors

### CODE REVIEW

**Comment by reaperhulk** [comment]
We need to add the provenance of these vectors to `test-vectors.rst`

**Comment by Ayrx** [comment]
@reaperhulk Is the short 2 liner I added enough considering the fact that it wasn't *us* who generated the vectors? Or do you want me to write a verification script like for the vectors we generate?

**Comment by reaperhulk** [comment]
No, I mean we require all test vectors added to have information about where they came from on this page: https://cryptography.io/en/latest/development/test-vectors/

In this particular case you'd want to link to where you got them from. The discussion isn't sufficient -- we need a link to them in the openssl commits. :)

**Comment by Ayrx** [comment]
Done. Hopefully I didn't mess up the RST linking syntax. ;)

File: docs/development/test-vectors.rst [hunk 1]
```diff
@@ -332,6 +332,14 @@ Hashes
 * SHA2 (224, 256, 384, 512) from `NIST CAVP`_.
 * Whirlpool from the `Whirlpool website`_.
 
+Custom Hashes
+~~~~~~~~~~~~~
+
+* Blake2 - Generated as part of the OpenSSL commit that added Blake2
+  functionality. Can be found in the OpenSSL repository under
+  ``test/evptests.txt``. Verified against the reference implementation of
```
Comment (reaperhulk) [inline docs/development/test-vectors.rst]: Should be 
```
`test/evptests.txt`_
```

File: docs/development/test-vectors.rst [hunk 2]
```diff
@@ -332,6 +332,14 @@ Hashes
 * SHA2 (224, 256, 384, 512) from `NIST CAVP`_.
 * Whirlpool from the `Whirlpool website`_.
 
+Custom Hashes
```
Comment (reaperhulk) [inline docs/development/test-vectors.rst]: This section isn't required, you can just add a line after Whirlpool above that says


```
* Blake2s and Blake2b from OpenSSL `test/evptests.txt`_.
```

**Comment by Ayrx** [comment]
@reaperhulk Done!

**Comment by reaperhulk** [comment]
Thanks @Ayrx. Vectors verified. Will merge when tests pass.

**Comment by reaperhulk** [comment]
So GitHub doesn't want to let me merge this. Might be a bug with required checks. Would you mind rebasing this to see if that fixes it?

**Comment by Ayrx** [comment]
@reaperhulk Is it ok now?

**Comment by reaperhulk** [comment]
yep :)

**Pull request merged**